tensorpipe/core/buffer.h

```diff
@@ -59,4 +60,21 @@ struct Buffer {
   };
 };
 
+template <typename TBuffer>
+TBuffer unwrap(Buffer);
```

Comment:
This is now used not only by the `PipeImpl` but also by the `ChannelBoilerplate`, hence it needs to be declared in a header.